internal/datanode/buffer.go

```diff
@@ -17,13 +17,220 @@
 package datanode
 
 import (
+	"container/heap"
 	"errors"
+	"fmt"
+	"github.com/milvus-io/milvus-proto/go-api/schemapb"
+	"github.com/milvus-io/milvus/internal/metrics"
 	"math"
+	"sync"
 
 	"github.com/milvus-io/milvus/internal/proto/datapb"
 	"github.com/milvus-io/milvus/internal/storage"
 )
 
```

Comment:
using buf manager to replace sync.map, which is too weak to maintain related data in the memory